homeassistant/components/honeywell/__init__.py

```diff
@@ -51,18 +49,29 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     username = config_entry.data[CONF_USERNAME]
     password = config_entry.data[CONF_PASSWORD]
 
-    client = await hass.async_add_executor_job(
-        get_somecomfort_client, username, password
+    client = AIOSomecomfort.AIOSomeComfort(
+        username, password, session=async_get_clientsession(hass)
     )
+    try:
+        await client.login()
+        await client.discover()
 
-    if client is None:
-        return False
-
+    except AIOSomecomfort.AuthError as ex:
+        # change to reauth flow in next PR that adds reauth flow.
```

Comment:
Please add these kind of comments as review comments instead of as code comments 👍 
```suggestion
```